widget/text.go

```diff
@@ -341,11 +341,55 @@ func splitLines(text []rune) [][2]int {
 	return append(lines, [2]int{low, length})
 }
 
+// newBinarySearch ccepts a function to measure line width, a slice of runes and a maximum line width
+// newBinarySearch returns a function that accepts the start and end rune index and returns the index of rune
+// located as close to the maximum line width as possible
+func newBinarySearch(measurer func([]rune) int, text []rune, maxWidth int) func(low int, maxHigh int) int {
+	return func(low int, maxHigh int) int {
+		if low >= maxHigh {
+		 	return low
+		}
+		if measurer(text[low:maxHigh]) <= maxWidth {
+			return maxHigh
+		}
+		high := low
+		delta := maxHigh - low
+		for delta > 0 {
+			delta /= 2
+			if measurer(text[low:high+delta]) < maxWidth {
+				high += delta
+			}
+		}
+		for (high < maxHigh) && measurer(text[low:high+1]) <= maxWidth {
+			high++
+		}
+		return high
+	}
+}
+
+// findSpaceIndex accept a slice of runes and the last rune index
+// findSpaceIndex returns an index of space in the text or if not found the last rune index
```

Comment:
findSpaceIndex returns the index of the last space in the text, or fallback if there are no spaces